packages/web3-shared/base/src/utils/timestamp.ts

```diff
@@ -2,5 +2,6 @@ const ETHEREUM_GENESIS_BLOCK_DATE = 1262304000000 // 2010-01-01
 
 export function isValidTimestamp(time?: number) {
     if (!time) return false
+    if (time.toString().length === 10) return time * 1000 >= ETHEREUM_GENESIS_BLOCK_DATE
```

Comment:
We'd better preprocess the `time` by `* 1000` before invoking this method. Add this leads to the method being error-prone.